src/libcore/ptr/mod.rs

```diff
@@ -795,6 +801,7 @@ pub unsafe fn read_unaligned<T>(src: *const T) -> T {
 #[inline]
 #[stable(feature = "rust1", since = "1.0.0")]
 pub unsafe fn write<T>(dst: *mut T, src: T) {
+    debug_assert!(is_aligned_and_not_null(dst), "attempt to write to unaligned or null pointer");
```

Comment:
@eddyb you asked if this goes through `format_args` to explain the higher stack usage; I tried to find out but `assert!` is a built-in macro so I do not know how it expands.

Reply:
You can use `rustc --pretty=expanded` to find out, but from looking at MIR there shouldn't be any formatting overhead.